### PR TITLE
Fix a version text (Rewrite update check)

### PR DESCRIPTION
**Pull request by skitaoka**
https://github.com/opentoonz/opentoonz/pull/243

- Fix a version text (and its URL)
- use SIGNAL/SLOT macros for connect
- add a revision number to version string

This PR follows https://github.com/opentoonz/opentoonz/pull/286.

### CODE REVIEW

**Comment by skitaoka** [comment]
Jenkins

**Comment by skitaoka** [comment]
Jenkins

**Comment by skitaoka** [comment]
Jenkins

**Comment by skitaoka** [comment]
Jenkins

**Comment by skitaoka** [comment]
Jenkins

File: toonz/sources/common/trop/tresample.cpp [hunk 1]
```diff
@@ -2947,7 +2947,7 @@ TAffine aff_0, inv_0;
 	minmax(negradx_ - rad_x, negrady_ - rad_y, posradx_ + rad_x, posrady_ + rad_y, inv_0,
 		   negradu_, negradv_, posradu_, posradv_);
 
-	//free_nocalc (coln);
+	/////////////////////////////////////////////////////////
```
Comment (kogaki) [inline toonz/sources/common/trop/tresample.cpp]: Meanless comment ?

File: toonz/sources/toonz/main.cpp [hunk 2]
```diff
@@ -140,7 +141,7 @@ TEnv::StringVar EnvSoftwareCurrentFont("SoftwareCurrentFont", "Hervetica");
 TEnv::IntVar EnvSoftwareCurrentFontSize("SoftwareCurrentFontSize", 12);
 TEnv::StringVar EnvSoftwareCurrentFontWeight("SoftwareCurrentFontWeightIsBold", "Yes");
 
-const char *applicationFullName = "OpenToonz 1.0";
+const char *applicationFullName = "OpenToonz 1.0.2";
```
Comment (kogaki) [inline toonz/sources/toonz/main.cpp]: This can be constructed from other const vars.

File: toonz/sources/toonzqt/updatechecker.cpp [hunk 3]
```diff
@@ -1,227 +1,37 @@
...
 
-	int startIndex = qstr.indexOf("Startdate");
-	int endIndex = qstr.indexOf("Enddate");
+	// Convert the response from a QByteArray into a QString
+	QString candidateVersion = QString(reply->readAll()).trimmed();
 
-	if ((endIndex - startIndex - 9) <= 0) {
-		abort();
+	// TODO: Verify that the response was valid by ensuring we have a single line in the format x.x[.x]*
+	// For now, we can do a lazy version by ensuring we have at least a period
```
Comment (kogaki) [inline toonz/sources/toonzqt/updatechecker.cpp]: `For now, we just ensure that the response contains at least one period` ?

File: toonz/sources/toonz/main.cpp (hunk 2 above)
Comment (skitaoka) [inline toonz/sources/toonz/main.cpp]: This is for minimal modifications.

**Comment by kogaki** [comment]
merged. Thanks @ss23 and @skitaoka

**Pull request merged**